### ISSUE TITLE
System.Diagnostics.ProcessTests.ProcessTest.Process_MainModule test failure in CI

### ISSUE THREAD

**Issue by hpshelton** [dotnet/corefx#1499]
The [System.Diagnostics.ProcessTests.ProcessTest.Process_MainModule test](http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_debug_prtest/955/testReport/System.Diagnostics.ProcessTests/ProcessTest/Process_MainModule/) is failing from [Jenkins build 955](http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_debug_prtest/955/).

```
MESSAGE:
System.ComponentModel.Win32Exception : Only part of a ReadProcessMemory or WriteProcessMemory request was completed
+++++++++++++++++++
STACK TRACE:
at System.Diagnostics.NtProcessManager.GetModuleInfos(Int32 processId, Boolean firstModuleOnly)
in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs:line 407 
at System.Diagnostics.NtProcessManager.GetFirstModuleInfo(Int32 processId) in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs:line 299
at System.Diagnostics.Process.get_MainModule() in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Diagnostics.Process\src\System\Diagnostics\Process.Windows.cs:line 127 
at System.Diagnostics.ProcessTests.ProcessTest.Process_MainModule() 
in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Diagnostics.Process\tests\System.Diagnostics.Process.Tests\ProcessTest.cs:line 232
```

**Comment by ellismg** [comment]
I believe this can happen in the cases where the OS loader is loading a module while we are trying to enumerate them.  I am not sure what can be done in this case.  Perhaps try a few dozen times with a sleep if we get a Win32Exception?

**Comment by stephentoub** [comment]
Seems like we already do such a retry:
https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs#L389

**Comment by krwq** [comment]
I believe that Sleep might be too low. This is running on pretty slow CI so I believe Sleep(1) is gonna be up to 50ms which in total gives roughly 2.5s

**Comment by pallavit** [comment]
I don't think we should add sleep to the test. As @stephentoub pointed out we already have sleep in the implementation.
Also as per https://msdn.microsoft.com/en-us/library/windows/desktop/ms682631(v=vs.85).aspx EnumProcessModules is pretty much a best effort API and is not guaranteed to work (one of the failure scenarios is what we hit here). I think we should update the test to catch(Win32Exception) instead. Let me know if anyone thinks otherwise.

**Comment by stephentoub** [comment]
@pallavit, I've seen this fail more often recently, e.g. http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_release_prtest/1604/console.  I agree with your conclusion: if this is a valid outcome of the test, then we should handle it in the test rather than allowing an unactionable-failure to take out a build.  Ideally, though, we'd handle it in the test in a way that wouldn't mask other failures.

**Comment by Priya91** [comment]
This is now fixed with the recent changes, #2513
